From 4207e9d450a976baaa14835353d16116e3a15463 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 31 Mar 2025 11:14:14 -0700
Subject: [PATCH] XsubcontrollerDxe: Fix build with gcc-15

GCC-15 does not like the fact that the function signature does not match
when it is passed as a function pointer

Fixes

XusbControllerDxe.c:2403:19: error: passing argument 3 of 'gBS->CreateEvent' from incompatible pointer type [-Wincompatible-pointer-types]
XusbControllerDxe.c:2403:19: note: expected 'EFI_EVENT_NOTIFY' {aka 'void (*)(void *, void *)'} but argument is of type 'void (*)(UINT32)' {aka 'void (*)(unsigned int)'}

Upstream-Status: Submitted [https://github.com/NVIDIA/edk2-nvidia/pull/124]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Silicon/NVIDIA/Drivers/XusbControllerDxe/XusbControllerDxe.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Silicon/NVIDIA/Drivers/XusbControllerDxe/XusbControllerDxe.c b/Silicon/NVIDIA/Drivers/XusbControllerDxe/XusbControllerDxe.c
index c6116f2f..0951f1a1 100644
--- a/Silicon/NVIDIA/Drivers/XusbControllerDxe/XusbControllerDxe.c
+++ b/Silicon/NVIDIA/Drivers/XusbControllerDxe/XusbControllerDxe.c
@@ -2280,6 +2280,8 @@ fail:
 STATIC
 VOID
 XudcCheckInterrupts (
+  IN        VOID  *p,
+  IN        VOID  *q
   )
 {
   XudcPollForEvent (0x10UL);
